sec_certs/sample/cpe.py

```diff
@@ -8,6 +8,15 @@
 from sec_certs.utils import helpers
 
 
+class CPEConfiguration:
+    def __init__(self, platform: "CPE", cpes: list["CPE"]) -> None:
+        self.platform: "CPE" = platform
+        self.cpes: list["CPE"] = cpes
```

Comment:
Do we actually need to hold the whole `CPE` object here? 

In the function below, you run potentially costly `map()` operation every time we try to match something. If we stored only `uri` instead of the whole CPE, we could get rid of this operation. Would that be possible?

Reply:
Also, I'm undecided, but I'm thinking whether we should wrap all `CPE`s into this `CPEConfiguration` object, because technically a single CPE is a configuration as well. It would create some overhead, but semantically we would get rid of the schizophreny when we sometimes match `CPE` and sometimes `CPEConfiguration`. Not sure, what do you think?

Reply:
@adamjanovsky I can easily refactor the code to just store CPE URI, however, I wanted to create a factory method for this class that would check if any of the CPEs in `self.cpes` list is vulnerable. If there is any vulnerable CPE, I would create the `CPEConfiguration` object. In case there would be no vulnerable CPE, I would return `None` or throw some exception.

Reply:
In the end, my approach with the factory method is not needed anymore. I solved that with `parse_only_vulnerable_cpes` parameter in `_parse_nist_dict` function.